### PR TITLE
clean up handle web API errors

### LINKED ISSUE #20157: Missing info to not mark method with HttpGet
I happened to mark the action of an ErrorHandler class with an HttpGetAttribute which caused the action to not be executed when an exception was thrown.

It took me some time to figure out what the problem was. Afterwards I found the following paragraph in the docs for Handle Errors in Web Apps in the Fundamentals section:

> Don't mark the error handler action method with HTTP method attributes, such as HttpGet. Explicit verbs prevent some requests from reaching the method. Allow anonymous access to the method if unauthenticated users should see the error view.

Would have been very helpful to have that paragraph in the help for Web API error handling too.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3f6504cd-a059-c5d4-73fb-61222ef1c6e0
* Version Independent ID: 1b42352c-a34f-7703-aa3e-d7e3156bd0bd
* Content: [Handle errors in ASP.NET Core web APIs](https://docs.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-3.1)
* Content Source: [aspnetcore/web-api/handle-errors.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/web-api/handle-errors.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-webapi**
* GitHub Login: @pranavkm
* Microsoft Alias: **prkrishn**

### LINKED ISSUE #21119: Explain ControllerBase.Problem
Need reference to Problem and magic number clean up.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3f6504cd-a059-c5d4-73fb-61222ef1c6e0
* Version Independent ID: 1b42352c-a34f-7703-aa3e-d7e3156bd0bd
* Content: [Handle errors in ASP.NET Core web APIs](https://docs.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-5.0#exception-handler)
* Content Source: [aspnetcore/web-api/handle-errors.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/web-api/handle-errors.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-webapi**
* GitHub Login: @pranavkm
* Microsoft Alias: **prkrishn**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #21119
Fixes  #20157

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@serpent5  hold off, fixing a few more things.

**Comment by Rick-Anderson** [comment]
@serpent5 ready to review

File: aspnetcore/web-api/handle-errors.md [hunk 1]
```diff
@@ -217,7 +221,7 @@ The contents of the response can be modified from outside of the controller. In
 
     [!code-csharp[](handle-errors/samples/3.x/Filters/HttpResponseExceptionFilter.cs?name=snippet_HttpResponseExceptionFilter)]
 
-    In the preceding filter, the magic number 10 is subtracted from the maximum integer value. Subtracting this number allows other filters to run at the very end of the pipeline.
+    In the preceding filter, 10 is subtracted from the maximum integer value. Subtracting 10 allows other filters to run at the end of the pipeline.
```
Comment (serpent5) [inline aspnetcore/web-api/handle-errors.md:224]:
```suggestion
    The preceding filter specifies an `Order` of the maximum integer value minus 10. This allows other filters to run at the end of the pipeline.
```

**Review by serpent5 — APPROVED** [review]
LGTM, with just the one, small suggestion to consider.

**Pull request merged**